.github/workflows/tests.yml

```diff
@@ -231,17 +231,11 @@ jobs:
           for flag in nosnaptrim noscrub nobackfill norebalance norecover noscrub nodeep-scrub; do
               sudo microceph.ceph osd set $flag
           done
-          # Use ephemeral disk mounted on /mnt for ceph OSD.
-          # The block-devices plug doesn't allow accessing /dev/loopX devices so we make those same devices
-          # available under alternate names (/dev/sdiY) that are not used inside GitHub Action runners.
+          # Repurpose the ephemeral disk for ceph OSD.
           sudo swapoff /mnt/swapfile
-          sudo rm -f /mnt/swapfile
-          loop_file="/mnt/ceph-osd.img"
-          sudo fallocate -l 10G "${loop_file}"
-          loop_dev="$(sudo losetup --show --direct-io=on --nooverlap -f "${loop_file}")"
-          devInfo=($(sudo stat -c '%t %T' "${loop_dev}"))
-          sudo mknod -m 0660 /dev/sdia b 0x"${devInfo[0]}" 0x"${devInfo[1]}"
-          sudo microceph disk add --wipe /dev/sdia
+          ephemeral_disk="$(findmnt --noheadings --output SOURCE --target /mnt | sed 's/[0-9]\+$//')"
+          sudo umount /mnt
+          sudo microceph disk add --wipe "${ephemeral_disk}"
```

Comment:
@simondeziel oh did the microceph folks fix the apparmor issue that was preventing the use of partitions?

Reply:
@tomponline yeah, I too noticed the ephemeral disk was not always setup the same way. Sometimes `/mnt` was a `NTFS` partition, some other times it was `ext4`. Sometimes it would be `/dev/sdb1`, some other `/dev/sda1`. I've seen `NTFS` parts at least 2 times and it's likely not good to back a loop-dev. See `/dev/sdb1` in `blkid`'s output:

```
# From https://github.com/canonical/lxd/actions/runs/6177130891/job/16767605386
# but https://github.com/canonical/lxd/actions/runs/6177049637/job/16767324493 also was NTFS
...
+ swapon -s
Filename				Type		Size		Used		Priority
/mnt/swapfile                           file		4194300		0		-2
+ lsblk
NAME    MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
loop0     7:0    0 63.4M  1 loop /snap/core20/1974
loop1     7:1    0 73.9M  1 loop /snap/core22/864
loop2     7:2    0 53.3M  1 loop /snap/snapd/19457
loop3     7:3    0 90.2M  1 loop /snap/microceph/585
sda       8:0    0   86G  0 disk 
├─sda1    8:1    0 85.9G  0 part /
├─sda14   8:14   0    4M  0 part 
+ blkid
+ sudo swapoff /mnt/swapfile
└─sda15   8:15   0  106M  0 part /boot/efi
sdb       8:16   0   14G  0 disk 
└─sdb1    8:17   0   14G  0 part /mnt
/dev/sdb1: LABEL="Temporary Storage" BLOCK_SIZE="512" UUID="24E0FC11E0FBE748" TYPE="ntfs" PARTUUID="f05c5304-01"
/dev/sda15: LABEL_FATBOOT="UEFI" LABEL="UEFI" UUID="D659-5A9F" BLOCK_SIZE="512" TYPE="vfat" PARTUUID="59204cef-8bed-4e09-876e-f620523d8884"
/dev/sda1: LABEL="cloudimg-rootfs" UUID="d4004e88-d528-486d-b994-90c70f9322d4" BLOCK_SIZE="4096" TYPE="ext4" PARTUUID="886ad3da-e4ec-442e-b223-03e615c2b78d"
/dev/loop1: TYPE="squashfs"
/dev/loop2: TYPE="squashfs"
/dev/loop0: TYPE="squashfs"
++ findmnt --noheadings --output SOURCE --target /mnt
++ sed 's/[0-9]\+$//'
+ ephemeral_disk=/dev/sdb
+ sudo umount /mnt
+ sudo microceph disk add --wipe /dev/sdb
```

As for the part vs whole disk, I didn't think of Apparmor but it could well be. My only thought for Apparmor was the immediate benefit of no longer needing to masquerade a loop device as `sdia` :) That is moot now that we use `/dev/sda` or `/dev/sdb`.

Reply:
Lets see if it works, previously microceph's apparmor profile prevented using partitions as disks, but hopefully they fixed that.

Reply:
I don't think they fixed it:

```
+ sudo microceph disk add --wipe /dev/sda1
Error: Failed adding new disk: Failed to wipe the device: Failed to run: dd if=/dev/zero of=/dev/disk/by-id/wwn-0x600224800af2a3230d891ce3c7debbb5-part1 bs=4M count=10 status=none: exit status 1 (dd: failed to open '/dev/disk/by-id/wwn-0x600224800af2a3230d891ce3c7debbb5-part1': Permission denied)
```